src/common.cpp

```diff
@@ -1071,7 +1071,7 @@ size_t read_unquoted_escape(const wchar_t *input, wcstring *result, bool allow_i
             const wchar_t sequence_char = input[in_pos++];
             if (sequence_char >= L'a' && sequence_char <= (L'a' + 32)) {
                 result_char_or_none = sequence_char - L'a' + 1;
-            } else if (sequence_char >= L'A' && sequence_char <= (L'A' + 32)) {
+            } else if (sequence_char >= L'@' && sequence_char <= (L'A' + 32)) {
```

Comment:
This was wrong before your change and it's now only slightly less wrong :smile: The current implementation is also needlessly complicated. if/else if blocks can be replaced by a single if to verify the value is in the range 0x20 <= c < 0x1F followed by a statement that simply masks off the low five bits of the char.